core/src/main/scala/org/apache/spark/status/storeTypes.scala

```diff
@@ -73,6 +73,8 @@ private[spark] class JobDataWrapper(
   @JsonIgnore @KVIndex
   private def id: Int = info.jobId
 
+  @JsonIgnore @KVIndex("completionTime")
+  private def completionTime: Long = info.completionTime.map(_.getTime).getOrElse(Long.MaxValue)
```

Comment:
This is fine, but if you want you can probably replace the filters in the listener by setting this to `-1` for running jobs / stages / others, and starting iteration at "0".